src/nvim/charset.c

```diff
@@ -1676,26 +1676,37 @@ char_u* skiptowhite_esc(char_u *p) {
   return p;
 }
 
-/// Getdigits: Get a number from a string and skip over it.
+/// Get a number from a string and skip over it.
 ///
-/// Note: the argument is a pointer to a char_u pointer!
+/// @param[out]  pp  A pointer to a pointer to char_u.
+///                  It will be advanced past the read number.
 ///
-/// @param pp
+/// @return Number read from the string.
+intmax_t get_digits(char_u **pp)
+{
+  intmax_t number = strtoimax((char *)*pp, (char **)pp, 10);
+  assert(errno != ERANGE);
+  return number;
+}
+
+/// Get an int number from a string.
 ///
-/// @return Number from the string.
-long getdigits(char_u **pp)
+/// A get_digits wrapper restricted to int values.
+int get_int_digits(char_u **pp)
 {
-  char_u *p = *pp;
-  long retval = atol((char *)p);
+  intmax_t number = get_digits(pp);
+  assert(number >= INT_MIN && number <= INT_MAX);
+  return (int)number;
+}
 
-  if (*p == '-') {
-    // skip negative sign
-    ++p;
-  }
-  // skip to next non-digit
-  p = skipdigits(p);
-  *pp = p;
-  return retval;
+/// Get a long number from a string.
+///
+/// A get_digits wrapper restricted to long values.
+long get_long_digits(char_u **pp)
```

Comment:
I did in fact doubt about that when initially doing it.
In the end, I opted for the more "natural speech" option, but it was probably the wrong option.
You're right the other approach is more discoverable.
I'll rename them.